tests/kafka/message_adapter_test.py

```diff
@@ -137,37 +293,418 @@ def message_with_schema(schema: str) -> KafkaMessage:
 
 
-def test_RouteBySchemaAdapter_calls_adapter_based_on_route() -> None:
-    class Adapter:
-        def __init__(self, value: str):
-            self._value = value
+class TestBeamlimeConfigMessageAdapter:
+    def test_adapter(self) -> None:
+        key = b'my_source/my_service/my_key'
+        encoded = json.dumps('my_value').encode('utf-8')
+        message = FakeKafkaMessage(
+            key=key, value=encoded, topic="dummy_beamlime_commands"
+        )
+        adapter = BeamlimeConfigMessageAdapter()
+        adapted_message = adapter.adapt(message)
+        # So it gets routed to config handler
+        assert adapted_message.stream == CONFIG_STREAM_ID
+        assert adapted_message.value == RawConfigItem(key=key, value=encoded)
+
+
+class TestErrorHandling:
+    """Tests for handling malformed or corrupt Kafka messages."""
+
+    def setup_method(self):
+        self.fake_logger = self._create_fake_logger()
+
+    def _create_fake_logger(self):
+        class FakeLogger:
+            def __init__(self):
+                self.warning_calls = []
+                self.exception_calls = []
+                self.info_calls = []
+
+            def warning(self, message, *args, **kwargs):
+                self.warning_calls.append((message, args, kwargs))
+
+            def exception(self, message, *args, **kwargs):
+                self.exception_calls.append((message, args, kwargs))
+
+            def info(self, message, *args, **kwargs):
+                self.info_calls.append((message, args, kwargs))
+
+        return FakeLogger()
+
+    def test_corrupt_ev44_message(self, monkeypatch):
+        """Test handling of corrupt ev44 message."""
+        # Create a message that has the ev44 schema identifier but corrupt content
+        corrupt_ev44 = bytearray(make_serialized_ev44())
+        # Corrupt the data after the schema identifier
+        if len(corrupt_ev44) > 12:
+            corrupt_ev44[12:] = b'corrupt' * 10
+
+        class CorruptEv44Source(MessageSource[KafkaMessage]):
+            def get_messages(self):
+                return [FakeKafkaMessage(value=bytes(corrupt_ev44), topic="monitors")]
+
+        def mock_deserialize(*args, **kwargs):
+            raise ValueError("Failed to deserialize corrupt ev44 data")
+
+        monkeypatch.setattr(
+            "streaming_data_types.eventdata_ev44.deserialise_ev44", mock_deserialize
+        )
+
+        source = AdaptingMessageSource(
+            source=CorruptEv44Source(),
+            adapter=KafkaToEv44Adapter(stream_kind=StreamKind.MONITOR_EVENTS),
+            logger=self.fake_logger,
+        )
+
+        # The exception should be caught and logged, then re-raised
+        with pytest.raises(ValueError, match="Failed to deserialize corrupt ev44 data"):
+            source.get_messages()
+
+        assert len(self.fake_logger.exception_calls) == 1
+        assert (
+            "error adapting message" in self.fake_logger.exception_calls[0][0].lower()
```

Comment:
```suggestion
            "error adapting message" in self.fake_logger.exception_calls[-1][0].lower()
```

Should we rather use [-1] index because it should be the latest one?
Also it won't break even if we insert more log related tests above.

Reply:
Ah sorry now I understood. The fake logger is set up before every method. Feel free to ignore this suggestion.